src/components/BounceRuleDetailed/index.jsx

```diff
@@ -1,61 +1,531 @@
+          </TableRow>
+        </TableBody>
+      </Table>
+    </div>
+  </div>
+);
+
+const DetailContainerEditable = ({ currentRule, onChangeRule }) => (
+  <div className="detail-container detail-container-editable card ">
+    <div className="editable">
+      <i className="sg-icon sg-icon-editor-design" />
+    </div>
+    <div className="description-info">
+      <Table className="table-fixed">
+        <TableBody>
+          <TableRow>
+            <TableCell className="description-cell">
+              <strong>Description</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "description")}
+                value={currentRule.description}
+                type="text"
+                label={currentRule.description}
+              />
+            </TableCell>
+          </TableRow>
+        </TableBody>
+      </Table>
+    </div>
+    <div className="detail-info">
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell>
+              <strong>Bounce ID</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "id")}
+                value={currentRule.id}
+                type="text"
+                label={currentRule.id}
+              />
+            </TableCell>
+          </TableRow>
+          <TableRow>
+            <TableCell>
+              <strong>Response Code</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "response_code")}
+                value={currentRule.response_code}
+                type="text"
+                label={currentRule.response_code}
+              />
+            </TableCell>
+          </TableRow>
+          <TableRow>
+            <TableCell>
+              <strong>Enhanced Code</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "enhanced_code")}
+                value={currentRule.enhanced_code}
+                type="text"
+                label={currentRule.enhanced_code}
+              />
+            </TableCell>
+          </TableRow>
+        </TableBody>
+      </Table>
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell>
+              <strong>Regex</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "regex")}
+                value={currentRule.regex}
+                type="text"
+                label={currentRule.regex}
+              />
+            </TableCell>
+          </TableRow>
+          <TableRow>
+            <TableCell>
+              <strong>Priority</strong>
+            </TableCell>
+            <TableCell>
+              <TextInput
+                onChange={e => onChangeRule(e, "priority")}
```

Comment:
We can handle the onChangeRule another way such that it's not always declaring a new function on each of these spots in every re-render. We can put an id prop that matches the state name you're trying to change on the TextInput and then in the onChange it will be const { id, value } = e.currentTarget; and then setting the new state to be something like { [id]: value } so the function is only passed down and no extra functions are getting created